### ISSUE TITLE
ssh -J <teleport-proxy> fails with tls routing enabled

### ISSUE THREAD

**Issue by Joerger** [gravitational/teleport#13833]
Expected behavior:

According to the test plan, `ssh -J <teleport-proxy>` should work. 

Current behavior:

When the Teleport proxy receives the ssh request, it denies it with the log `[MX:PROXY:] Closing SSH connection: SSH listener is disabled. multiplexer/multiplexer.go:241`.

Is it possible for us to multiplex SSH proxy connections on the proxy listener? Or do we need to update the testplan to only expect this with tls routing disabled?

Bug details:
- Teleport version - All versions with tls routing
- Recreation steps - `ssh -p 3022 -J proxy.example.com:3080 server01`

**Comment by r0mant** [comment]
Hmm I'm pretty sure we fixed that before and it did work, @smallinsky could you check it out please?

**Comment by smallinsky** [comment]
@r0mant  
> Hmm I'm pretty sure we fixed that before and it did work

I think that TLS Routing web port never handled raw SSH protocol and this is way we introduced the `tsh proxy ssh` command so `-J proxy.example.com:3080` won't work and instead the `ProxyCommand  tsh proxy` should be used. 

@Joerger 
> Is it possible for us to multiplex SSH proxy connections on the proxy listener? Or do we need to update the testplan to only expect this with TLS routing disabled?

IFAK we never supported multiplexing web port with ssh proxy (we only allowed to multiplex web port with reverse tunnel port) and the following configuration proxy configuration is invalid: 


```
  proxy_listener_mode: "separate"
...
proxy_service:
  listen_addr: 0.0.0.0:3080
  web_listen_addr: 0.0.0.0:3080
```

and will result in `listen tcp 0.0.0.0:3080: bind: address already in use`. 

I think that the `teleport-proxy` reference in `ssh -J <teleport-proxy>` command means teleport proxy ssh port not web port.
